methodological 0.10.0 not error compilation on repl

This is similar to the first comment ion the following issue but yet id differs:
https://github.com/camsaul/methodical/issues/33

I am trying to run methodical 0.10.0-alpha with java 1.8 and Clojure 1.10.0 and I get:

error in process sentinel: Could not start nREPL server: Compiling af-river.api.http
Unexpected error macroexpanding p.types/deftype+ at (methodical/impl/cache/simple.clj:9:1).
Exception in thread "main" Unexpected error macroexpanding p.types/deftype+ at (methodical/impl/cache/simple.clj:9:1).
	at clojure.lang.Compiler.macroexpand1(Compiler.java:7018)
	at clojure.lang.Compiler.macroexpand(Compiler.java:7074)
	at clojure.lang.Compiler.compile1(Compiler.java:7715)
	at clojure.lang.Compiler.compile(Compiler.java:7797)
	at clojure.lang.RT.compile(RT.java:415)
	at clojure.lang.RT.load(RT.java:461)
	at clojure.lang.RT.load(RT.java:428)
	at clojure.core$load$fn__6824.invoke(core.clj:6126)
	at clojure.core$load.invokeStatic(core.clj:6125)
	at clojure.core$load.doInvoke(core.clj:6109)
	at clojure.lang.RestFn.invoke(RestFn.java:408)
	at clojure.core$load_one.invokeStatic(core.clj:5908)
	at clojure.core$load_one.invoke(core.clj:5903)
	at clojure.core$load_lib$fn__6765.invoke(core.clj:5948)
	at clojure.core$load_lib.invokeStatic(core.clj:5947)
	at clojure.core$load_lib.doInvoke(core.clj:5928)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$load_libs.invokeStatic(core.clj:5989)
	at clojure.core$load_libs.doInvoke(core.clj:5969)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$require.invokeStatic(core.clj:6007)
	at clojure.core$require.doInvoke(core.clj:6007)
	at clojure.lang.RestFn.invoke(RestFn.java:512)
	at methodical.impl$loading__6706__auto____389.invoke(impl.clj:1)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.lang.Compiler$InvokeExpr.eval(Compiler.java:3702)
	at clojure.lang.Compiler.compile1(Compiler.java:7730)
	at clojure.lang.Compiler.compile1(Compiler.java:7720)
	at clojure.lang.Compiler.compile(Compiler.java:7797)
	at clojure.lang.RT.compile(RT.java:415)
	at clojure.lang.RT.load(RT.java:461)
	at clojure.lang.RT.load(RT.java:428)
	at clojure.core$load$fn__6824.invoke(core.clj:6126)
	at clojure.core$load.invokeStatic(core.clj:6125)
	at clojure.core$load.doInvoke(core.clj:6109)
	at clojure.lang.RestFn.invoke(RestFn.java:408)
	at clojure.core$load_one.invokeStatic(core.clj:5908)
	at clojure.core$load_one.invoke(core.clj:5903)
	at clojure.core$load_lib$fn__6765.invoke(core.clj:5948)
	at clojure.core$load_lib.invokeStatic(core.clj:5947)
	at clojure.core$load_lib.doInvoke(core.clj:5928)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$load_libs.invokeStatic(core.clj:5989)
	at clojure.core$load_libs.doInvoke(core.clj:5969)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$require.invokeStatic(core.clj:6007)
	at clojure.core$require.doInvoke(core.clj:6007)
	at clojure.lang.RestFn.invoke(RestFn.java:421)
	at methodical.core$loading__6706__auto____387.invoke(core.clj:1)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.lang.Compiler$InvokeExpr.eval(Compiler.java:3702)
	at clojure.lang.Compiler.compile1(Compiler.java:7730)
	at clojure.lang.Compiler.compile1(Compiler.java:7720)
	at clojure.lang.Compiler.compile(Compiler.java:7797)
	at clojure.lang.RT.compile(RT.java:415)
	at clojure.lang.RT.load(RT.java:461)
	at clojure.lang.RT.load(RT.java:428)
	at clojure.core$load$fn__6824.invoke(core.clj:6126)
	at clojure.core$load.invokeStatic(core.clj:6125)
	at clojure.core$load.doInvoke(core.clj:6109)
	at clojure.lang.RestFn.invoke(RestFn.java:408)
	at clojure.core$load_one.invokeStatic(core.clj:5908)
	at clojure.core$load_one.invoke(core.clj:5903)
	at clojure.core$load_lib$fn__6765.invoke(core.clj:5948)
	at clojure.core$load_lib.invokeStatic(core.clj:5947)
	at clojure.core$load_lib.doInvoke(core.clj:5928)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$load_libs.invokeStatic(core.clj:5985)
	at clojure.core$load_libs.doInvoke(core.clj:5969)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$require.invokeStatic(core.clj:6007)
	at clojure.core$require.doInvoke(core.clj:6007)
	at clojure.lang.RestFn.invoke(RestFn.java:408)
	at af_river.block$loading__6706__auto____385.invoke(block.clj:1)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.lang.Compiler$InvokeExpr.eval(Compiler.java:3702)
	at clojure.lang.Compiler.compile1(Compiler.java:7730)
	at clojure.lang.Compiler.compile1(Compiler.java:7720)
	at clojure.lang.Compiler.compile(Compiler.java:7797)
	at clojure.lang.RT.compile(RT.java:415)
	at clojure.lang.RT.load(RT.java:461)
	at clojure.lang.RT.load(RT.java:428)
	at clojure.core$load$fn__6824.invoke(core.clj:6126)
	at clojure.core$load.invokeStatic(core.clj:6125)
	at clojure.core$load.doInvoke(core.clj:6109)
	at clojure.lang.RestFn.invoke(RestFn.java:408)
	at clojure.core$load_one.invokeStatic(core.clj:5908)
	at clojure.core$load_one.invoke(core.clj:5903)
	at clojure.core$load_lib$fn__6765.invoke(core.clj:5948)
	at clojure.core$load_lib.invokeStatic(core.clj:5947)
	at clojure.core$load_lib.doInvoke(core.clj:5928)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$load_libs.invokeStatic(core.clj:5985)
	at clojure.core$load_libs.doInvoke(core.clj:5969)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$require.invokeStatic(core.clj:6007)
	at clojure.core$require.doInvoke(core.clj:6007)
	at clojure.lang.RestFn.invoke(RestFn.java:619)
	at af_river.schema$loading__6706__auto____383.invoke(schema.clj:1)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.lang.Compiler$InvokeExpr.eval(Compiler.java:3702)
	at clojure.lang.Compiler.compile1(Compiler.java:7730)
	at clojure.lang.Compiler.compile1(Compiler.java:7720)
	at clojure.lang.Compiler.compile(Compiler.java:7797)
	at clojure.lang.RT.compile(RT.java:415)
	at clojure.lang.RT.load(RT.java:461)
	at clojure.lang.RT.load(RT.java:428)
	at clojure.core$load$fn__6824.invoke(core.clj:6126)
	at clojure.core$load.invokeStatic(core.clj:6125)
	at clojure.core$load.doInvoke(core.clj:6109)
	at clojure.lang.RestFn.invoke(RestFn.java:408)
	at clojure.core$load_one.invokeStatic(core.clj:5908)
	at clojure.core$load_one.invoke(core.clj:5903)
	at clojure.core$load_lib$fn__6765.invoke(core.clj:5948)
	at clojure.core$load_lib.invokeStatic(core.clj:5947)
	at clojure.core$load_lib.doInvoke(core.clj:5928)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$load_libs.invokeStatic(core.clj:5985)
	at clojure.core$load_libs.doInvoke(core.clj:5969)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$require.invokeStatic(core.clj:6007)
	at clojure.core$require.doInvoke(core.clj:6007)
	at clojure.lang.RestFn.invoke(RestFn.java:1523)
	at af_river.domain.lookup$loading__6706__auto____178.invoke(lookup.clj:1)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.lang.Compiler$InvokeExpr.eval(Compiler.java:3702)
	at clojure.lang.Compiler.compile1(Compiler.java:7730)
	at clojure.lang.Compiler.compile1(Compiler.java:7720)
	at clojure.lang.Compiler.compile(Compiler.java:7797)
	at clojure.lang.RT.compile(RT.java:415)
	at clojure.lang.RT.load(RT.java:461)
	at clojure.lang.RT.load(RT.java:428)
	at clojure.core$load$fn__6824.invoke(core.clj:6126)
	at clojure.core$load.invokeStatic(core.clj:6125)
	at clojure.core$load.doInvoke(core.clj:6109)
	at clojure.lang.RestFn.invoke(RestFn.java:408)
	at clojure.core$load_one.invokeStatic(core.clj:5908)
	at clojure.core$load_one.invoke(core.clj:5903)
	at clojure.core$load_lib$fn__6765.invoke(core.clj:5948)
	at clojure.core$load_lib.invokeStatic(core.clj:5947)
	at clojure.core$load_lib.doInvoke(core.clj:5928)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$load_libs.invokeStatic(core.clj:5985)
	at clojure.core$load_libs.doInvoke(core.clj:5969)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$require.invokeStatic(core.clj:6007)
	at clojure.core$require.doInvoke(core.clj:6007)
	at clojure.lang.RestFn.invoke(RestFn.java:3894)
	at af_river.api.http$loading__6706__auto____171.invoke(http.clj:1)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.lang.Compiler$InvokeExpr.eval(Compiler.java:3702)
	at clojure.lang.Compiler.compile1(Compiler.java:7730)
	at clojure.lang.Compiler.compile1(Compiler.java:7720)
	at clojure.lang.Compiler.compile(Compiler.java:7797)
	at clojure.lang.RT.compile(RT.java:415)
	at clojure.lang.RT.load(RT.java:461)
	at clojure.lang.RT.load(RT.java:428)
	at clojure.core$load$fn__6824.invoke(core.clj:6126)
	at clojure.core$load.invokeStatic(core.clj:6125)
	at clojure.core$load.doInvoke(core.clj:6109)
	at clojure.lang.RestFn.invoke(RestFn.java:408)
	at clojure.core$load_one.invokeStatic(core.clj:5908)
	at clojure.core$compile$fn__6829.invoke(core.clj:6136)
	at clojure.core$compile.invokeStatic(core.clj:6136)
	at clojure.core$compile.invoke(core.clj:6128)
	at user$eval155$fn__164.invoke(form-init5573748726313510021.clj:1)
	at user$eval155.invokeStatic(form-init5573748726313510021.clj:1)
	at user$eval155.invoke(form-init5573748726313510021.clj:1)
	at clojure.lang.Compiler.eval(Compiler.java:7176)
	at clojure.lang.Compiler.eval(Compiler.java:7166)
	at clojure.lang.Compiler.eval(Compiler.java:7166)
	at clojure.lang.Compiler.load(Compiler.java:7635)
	at clojure.lang.Compiler.loadFile(Compiler.java:7573)
	at clojure.main$load_script.invokeStatic(main.clj:452)
	at clojure.main$init_opt.invokeStatic(main.clj:454)
	at clojure.main$init_opt.invoke(main.clj:454)
	at clojure.main$initialize.invokeStatic(main.clj:485)
	at clojure.main$null_opt.invokeStatic(main.clj:519)
	at clojure.main$null_opt.invoke(main.clj:516)
	at clojure.main$main.invokeStatic(main.clj:598)
	at clojure.main$main.doInvoke(main.clj:561)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.lang.Var.applyTo(Var.java:705)
	at clojure.main.main(main.java:37)
Caused by: java.lang.NoClassDefFoundError: clojure/core$seq_QMARK___5406
	at potemkin.types$clean_deftype.invokeStatic(types.clj:72)
	at potemkin.types$deftype_PLUS_.invokeStatic(types.clj:291)
	at potemkin.types$deftype_PLUS_.doInvoke(types.clj:291)
	at clojure.lang.RestFn.applyTo(RestFn.java:151)
	at clojure.lang.Var.applyTo(Var.java:705)
	at clojure.lang.Compiler.macroexpand1(Compiler.java:6992)
	... 201 more
Caused by: java.lang.ClassNotFoundException: clojure.core$seq_QMARK___5406
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 207 more
Compilation failed: Subprocess failed

--- a/project.clj
+++ b/project.clj
@@ -28,8 +28,6 @@
   [[pretty "1.0.4"]
    [potemkin "0.4.5"]]
 
-  :aot [methodical.interface methodical.impl.standard]
-
   :jvm-opts ["-Dclojure.compiler.direct-linking=true"]
 
   :profiles
@@ -110,7 +108,8 @@
    {:plugins               [[lein-check-namespace-decls "1.0.2"
                              :exclusions [org.clojure/clojure]]]
     :source-paths          ["test"]
-    :check-namespace-decls {:prefix-rewriting true}}
+    :check-namespace-decls {:prefix-rewriting true
+                            :prune-ns-form    false}}
 
    :docstring-checker
    {:plugins

--- a/src/methodical/impl.clj
+++ b/src/methodical/impl.clj
@@ -18,7 +18,8 @@
             [methodical.impl.multifn
              [cached :as multifn.cached]
              [standard :as multifn.standard]]
-            [methodical.impl.standard :as impl.standard])
+            [methodical.impl.standard :as impl.standard]
+            methodical.interface)
   (:import methodical.impl.standard.StandardMultiFn
            [methodical.interface Cache Dispatcher MethodCombination MethodTable MultiFnImpl]))
 

--- a/src/methodical/impl/cache/simple.clj
+++ b/src/methodical/impl/cache/simple.clj
@@ -2,7 +2,8 @@
   "A basic, dumb cache. `SimpleCache` stores cached methods in a simple map of dispatch-value -> effective method; it
   offers no facilities to deduplicate identical methods for the same dispatch value. This behaves similarly to the
   caching mechanism in vanilla Clojure."
-  (:require [potemkin.types :as p.types]
+  (:require methodical.interface
+            [potemkin.types :as p.types]
             [pretty.core :refer [PrettyPrintable]])
   (:import methodical.interface.Cache))
 

--- a/src/methodical/impl/combo/clojure.clj
+++ b/src/methodical/impl/combo/clojure.clj
@@ -1,7 +1,8 @@
 (ns methodical.impl.combo.clojure
   "Simple method combination strategy that mimics the way vanilla Clojure multimethods combine methods; that is, to say,
   not at all. Like vanilla Clojure multimethods, this method combination only supports primary methods."
-  (:require [potemkin.types :as p.types]
+  (:require methodical.interface
+            [potemkin.types :as p.types]
             [pretty.core :refer [PrettyPrintable]])
   (:import methodical.interface.MethodCombination))
 

--- a/src/methodical/impl/combo/clos.clj
+++ b/src/methodical/impl/combo/clos.clj
@@ -4,6 +4,7 @@
   are ignored. Primary methods and around methods get an implicit `next-method` arg (see Methodical dox for more on
   what this means)."
   (:require [methodical.impl.combo.common :as combo.common]
+            methodical.interface
             [potemkin.types :as p.types]
             [pretty.core :refer [PrettyPrintable]])
   (:import methodical.interface.MethodCombination))

--- a/src/methodical/impl/combo/operator.clj
+++ b/src/methodical/impl/combo/operator.clj
@@ -38,6 +38,7 @@
       ...)"
   (:refer-clojure :exclude [methods])
   (:require [methodical.impl.combo.common :as combo.common]
+            methodical.interface
             [potemkin.types :as p.types]
             [pretty.core :refer [PrettyPrintable]])
   (:import methodical.interface.MethodCombination))

--- a/src/methodical/impl/combo/threaded.clj
+++ b/src/methodical/impl/combo/threaded.clj
@@ -1,6 +1,7 @@
 (ns methodical.impl.combo.threaded
   (:refer-clojure :exclude [methods])
   (:require [methodical.impl.combo.common :as combo.common]
+            methodical.interface
             [potemkin.types :as p.types]
             [pretty.core :refer [PrettyPrintable]])
   (:import methodical.interface.MethodCombination))

--- a/src/methodical/impl/method_table/clojure.clj
+++ b/src/methodical/impl/method_table/clojure.clj
@@ -1,5 +1,6 @@
 (ns methodical.impl.method-table.clojure
-  (:require [potemkin.types :as p.types]
+  (:require methodical.interface
+            [potemkin.types :as p.types]
             [pretty.core :refer [PrettyPrintable]])
   (:import methodical.interface.MethodTable))
 

--- a/src/methodical/impl/method_table/standard.clj
+++ b/src/methodical/impl/method_table/standard.clj
@@ -1,5 +1,6 @@
 (ns methodical.impl.method-table.standard
-  (:require [potemkin.types :as p.types]
+  (:require methodical.interface
+            [potemkin.types :as p.types]
             [pretty.core :refer [PrettyPrintable]])
   (:import methodical.interface.MethodTable))
 

--- a/test/methodical/impl/dispatcher/multi_default_test.clj
+++ b/test/methodical/impl/dispatcher/multi_default_test.clj
@@ -1,6 +1,6 @@
 (ns methodical.impl.dispatcher.multi-default-test
   (:require [clojure.test :as t]
-            [methodical.core :as m]
+            [methodical [core :as m] interface]
             [methodical.impl.dispatcher.multi-default :as multi-default])
   (:import methodical.interface.MethodTable))
 

